### PR TITLE
Fix semeio 23

### LINKED ISSUE #23: Calling scaling job from spearman job failing
Calling the scaling job from the spearman correlation job can fail with `"The script 'SpearmanCorrelationJob' caused an error while running: Empty dataset, all data has been filtered out' `, this is most likely caused by filtering both ensemble mean and towards observation mean in scaling job, and only ensemble mean in spearman correlation. This will happen more frequently because we can have clusters with a single data point, increasing the chance that the filtering will remove all the data. Should probably add a check to: 
```
def _run_scaling(facade, job_configs):
    for job in job_configs:
        scaling_job(facade, job)
```
to skip if the cluster size is 1 (or the scaling job should skip if only one member?)

### PR DESCRIPTION
**Pull request by jondequinor**
Fixes #23

### CODE REVIEW

**Comment by jondequinor** [comment]
> Think it would also be good to give the user feedback about why the observations were dropped.

It seems to me we're assuming it was because it was filtered out. But that has to imply that we knew it wasn't empty from the start. Are we sure the dataframe is empty due to filtering?

File: tests/jobs/spearman_correlation_job/test_integration.py [hunk 1]
```diff
@@ -54,4 +54,30 @@ def test_main_entry_point_gen_data(monkeypatch):
...
+    os.chdir(os.path.join("test_data"))
+
+    res_config = ResConfig("snake_oil.ert")
+
+    ert = EnKFMain(res_config)
+    facade = LibresFacade(ert)
+
+    try:
+        spearman.spearman_job(facade, 1.0, False)
+    except EmptyDatasetException:
```
[the remark this answers is not in the document]
Comment (jondequinor) [reply tests/jobs/spearman_correlation_job/test_integration.py]: It is not necessary, but two wins: 1) there's an assertion, communicating intent, and 2) we get an explicit failure and message if it fails.

**Comment by oyvindeide** [comment]
> > Think it would also be good to give the user feedback about why the observations were dropped.
> 
> It seems to me we're assuming it was because it was filtered out. But that has to imply that we knew it wasn't empty from the start. Are we sure the dataframe is empty due to filtering?
> 

Think I might have context switched on you in the middle of my comment, I meant that we should notify which clusters were dropped because they were consisting of a single data point, which has nothing to do with the filtering, it´s just that that data point is apparently not correlated to anything else.

**Comment by jondequinor** [comment]
> Think I might have context switched on you in the middle of my comment, I meant that we should notify which clusters were dropped because they were consisting of a single data point, which has nothing to do with the filtering, it´s just that that data point is apparently not correlated to anything else.

Okay, I totally misunderstood. Agree completely, will implement.

**Pull request merged**